File: go/api/arch/role_test.go
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/13
 */

package arch

import (
	"api-demo/base"
	"testing"
)

func TestRoleApi(t *testing.T) {
	api := new(RoleApiClient)
	api.ApiClient = &base.ApiClient{
		ApiKey: base.ApiKey,
		Host:   base.Host,
	}

	t.Run("RoleList", func(t *testing.T) {
		roleList, err := api.RoleList(make(map[string]interface{}))
		if err != nil {
			t.Error(err)
		} else {
			t.Log(base.JsonIndentString(roleList))
		}
	})
}

File: csharp/test/api/arch/Role.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("Role")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class RoleTest
{
    private static RoleApiClient api = new RoleApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void Test1()
    {
        var roleList = await api.roleList(new Dictionary<string, object>());
        Assert.True(roleList != null);
        Console.WriteLine(roleList);
    }
}